### PR TITLE
CLN: assorted

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pxd [hunk 1]
```diff
@@ -11,7 +11,6 @@ cdef bint is_any_td_scalar(object obj)
 cdef class _Timedelta(timedelta):
     cdef readonly:
         int64_t value      # nanoseconds
-        object freq        # frequency reference
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timedeltas.pxd:14]: i guess this is an API change, could be done via a deprecation

Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pxd:14]: if OK, i'll revert and then deprecate in a follow-up.

File: pandas/_libs/tslibs/dtypes.pyx [hunk 2]
```diff
@@ -165,7 +165,7 @@ class FreqGroup(Enum):
     FR_MS = c_FreqGroup.FR_MS
     FR_US = c_FreqGroup.FR_US
     FR_NS = c_FreqGroup.FR_NS
-    FR_UND = -c_FreqGroup.FR_UND  # undefined
+    FR_UND = c_FreqGroup.FR_UND  # undefined
```
Comment (mroeschke) [inline pandas/_libs/tslibs/dtypes.pyx:168]: Is `FR_UND`  essentially a dummy object so the sign doesn't matter?

File: pandas/_libs/tslibs/timedeltas.pxd (hunk 1 above)
Comment (mroeschke) [reply pandas/_libs/tslibs/timedeltas.pxd:14]: +1. Yeah appears this is never set.

IMO `is_populated` should be deprecated too and made private

File: pandas/_libs/tslibs/dtypes.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/dtypes.pyx:168]: mostly, but we could probably cook up a scenario where the mismatch between FreqGroup.FR_UND and c_FreqGroup.FR_UND messed something up

**Comment by jbrockmendel** [comment]
reverted Timedelta.freq removal, will do it The Right Way along with is_populated

**Review by mroeschke — APPROVED** [review]

**Pull request merged**